File: historical/milinda/src/org/openexi/proc/io/compression/Channel.java
package org.openexi.proc.io.compression;

abstract class Channel implements Comparable {
  
  final String name; 
  final String uri; 
  final int firstPos;
  
  int valueCount;
  protected final ChannelKeeper m_channelKeeper;
  
  Channel(String name, String uri, int firstPos, ChannelKeeper channelKeeper) {
    this.name = name;
    this.uri = uri;
    this.firstPos = firstPos;
    valueCount = 0;
    m_channelKeeper = channelKeeper;
  }

  public int compareTo(Object obj) {
    return firstPos - ((Channel)obj).firstPos;
  }
  
}

File: nagasenacs/src/Nagasena/Proc/Common/Channel.cs
using System;

namespace Nagasena.Proc.Common {

  /// <exclude/>
  public abstract class Channel : IComparable<Channel> {

    internal int firstPos;
    internal int blockNum;

    public int valueCount;

    protected internal Channel(int firstPos, int blockNum) {
      _reset(firstPos, blockNum);
    }

    private void _reset(int firstPos, int blockNum) {
      this.firstPos = firstPos;
      this.blockNum = blockNum;
      valueCount = 0;
    }

    public virtual void reset(int firstPos, int blockNum) {
      _reset(firstPos, blockNum);
    }

    public int CompareTo(Channel obj) {
      return firstPos - obj.firstPos;
    }

  }

}